tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.py

```diff
@@ -118,6 +120,8 @@ def task(event):
                             OUTPUT_DESTINATION_BUCKET_KEY: destination_bucket,
                         }
                     )
+            if len(files) == 0:
+                LOGGER.warning(f"All files for granule '{gran['granuleId']}' excluded.")
             gran["keys"] = files
```

Comment:
Not sure why we are adding empty array to the granule here if there are no files. Would it make sense to exclude the granule from the array? What are the implications if we do that? What if we just set a granule level failed state here for job tracking without processing any further? Thoughts?

Reply:
The next task in this chain, https://bugs.earthdata.nasa.gov/browse/ORCA-580, involves modifications to request_from_archive that key off of this state. Technically the status entries _could_ be created here, but it would require the sqs code to be copied here as well.